api/wikis/permissions.py

```diff
@@ -39,7 +42,23 @@ def has_object_permission(self, request, view, obj):
 class ExcludeWithdrawalsWikiVersion(permissions.BasePermission):
 
     def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, WikiVersion), 'obj must be a WikiVersion, got {}'.format(obj)
         node = obj.wiki_page.node
         if node and node.is_retracted:
             return False
         return True
+
+
+class IsEnabled(permissions.BasePermission):
```

Comment:
I'm not sure this makes sense as a permission class. We're not really checking if the client has permission as much as we are checking if the wiki exists at all.

How about putting this into the `get_object` logic?

Reply:
Yep, I agree. Done in d5d8b8b.